sympy/physics/control/lti.py

```diff
@@ -0,0 +1,181 @@
+from sympy import Basic, Mul, degree, Symbol, expand, cancel, Expr
+from sympy.core.numbers import Integer, Float
+from sympy.core.sympify import sympify, _sympify
+
+__all__ = ['TransferFunction',]
+
+
+class TransferFunction(Basic):
+
+    def __new__(cls, num, den, var):
+        num, den = _sympify(num), _sympify(den)
+
+        if not isinstance(var, Symbol):
+            raise TypeError("Variable input must be a Symbol.")
+        if den == 0:
+            raise ValueError("TransferFunction can't have a zero denominator.")
+
+        if (isinstance(num, Expr) and num.has(Symbol)) or isinstance(num, Integer):
+            if (isinstance(num, Expr) and num.has(Symbol)) or isinstance(num, Integer):
+                obj = super(TransferFunction, cls).__new__(cls, num, den, var)
+                obj._num = num
+                obj._den = den
+                obj._var = var
```

Comment:
@namannimmo10 Have you investigated using the Poly class internally? Here are the docs:

https://docs.sympy.org/latest/modules/polys/reference.html

I browsed it some this morning and see a number of methods that will likely be useful. What was your consideration on this?

Reply:
Yes, I tried storing the numerator and the denominator as `Poly` objects internally. i see it becomes totally illegible, given the fact that we are not keeping `num` and `den` in expanded form by default. Right now, it still gives a cleaner structure.. and if we want to change this, we have to make changes in all of the functionality.
``` python
>>> tf1 = TransferFunction((9 - s**3)*(s - 5)*(s + 1)*(s + 6) + (s - 5)*(s + 1)*(s + 3)**2, (9 - s**3)*(s + 1)*(s**4 - 2*s**2 + 1), s)
>>> tf1
TransferFunction(Poly(-s**6 - 2*s**5 + 30*s**4 + 41*s**3 - 2*s**2 - 327*s - 315, s, domain='ZZ'), Poly(-s**8 - s**7 + 2*s**6 + 11*s**5 + 8*s**4 - 19*s**3 - 18*s**2 + 9*s + 9, s, domain='ZZ'), s)
```
This is just one SISO transfer function, with a collection of TFs, `TFM = TransferFunctionMatrix([tf1, tf2])` this would not look good at all.

Reply:
That's just a printing issue, which can be adjusted. The issue is more about whether we need the functionality in the polynomial package and, if so, how much do we need? It may be true that we can get away with just using expressions and convert them to Poly for specific calcs if needed internally in the class.

Reply:
I understand. The polynomials package would be powerful in some use cases. 
> It may be true that we can get away with just using expressions and convert them to Poly for specific calcs if needed internally in the class.

If that can be done with some of its methods, can I proceed further now?

Reply:
Nothing is stopping you from proceeding. I'm asking you to answer my question. Should we use Poly? Why or why not?

Reply:
If I see https://github.com/sympy/sympy/pull/19390#issuecomment-635455769, I don't think that it is possible to have exponential with `Poly`, so it makes the design not a good option for the future.

And I think that we can always convert to Poly objects internally when it is needed for some computation, like it is often done for some sympy modules. 
Storing the poly objects in signature may only provide some efficient computation, but I may have some doubts about whether `Poly` is always more efficient than `Expr`, because expr can be in any forms like `(x - a)*(x - b)*(x - c)` such that the root computation can be trivial.

Reply:
I agree with @sylee957 here.
I think we should not use Poly internally by default. `Poly` objects often convert themselves into `Expr`. e.g.:
```python
>>> a = Poly(x**4 - 2*x**2 + 1)
>>> factor(a)
(x - 1)**2*(x + 1)**2
```
And we can always convert them to Poly objects for some specific computations whenever needed.

Reply:
I'm good with that. Thanks @sylee957 for the insight.